pkg/common/volume_operations_test.go

```diff
@@ -575,61 +617,75 @@ func TestVolumeOperationsImpl_ExpandVolume_DifferentStatuses(t *testing.T) {
+	assert.Nil(t, svc.k8sClient.CreateCR(testCtx, volumeCR.Spec.Id, &volumeCR))
 
 	for _, v := range [4]string{apiV1.Failed, apiV1.Removed, apiV1.Removing, apiV1.Creating} {
 		volumeCR.Spec.CSIStatus = v
-		err = svc.k8sClient.UpdateCR(testCtx, &volumeCR)
-		assert.Nil(t, err)
+		assert.Nil(t, svc.k8sClient.UpdateCR(testCtx, &volumeCR))
 		err := svc.ExpandVolume(testCtx, &volumeCR, capacity)
 		assert.NotNil(t, err)
 		assert.Equal(t, codes.FailedPrecondition, status.Code(err))
 	}
 
 	// Storage class is not lvg
 	volumeCR.Spec.CSIStatus = apiV1.Created
-	err = svc.k8sClient.UpdateCR(testCtx, &volumeCR)
-	err = svc.ExpandVolume(testCtx, &volumeCR, capacity)
-	assert.NotNil(t, err)
-	assert.Equal(t, codes.FailedPrecondition, status.Code(err))
+	volumeCR.ObjectMeta.ResourceVersion = ""
+	assert.NotNil(t, svc.k8sClient.UpdateCR(testCtx, &volumeCR))
+	if err := svc.ExpandVolume(testCtx, &volumeCR, capacity); err != nil {
+		assert.Equal(t, codes.FailedPrecondition, status.Code(err))
+	}
 
 	// Failed to get AC
+	volumeCR.ObjectMeta.ResourceVersion = ""
 	volumeCR.Spec.StorageClass = apiV1.StorageClassSystemLVG
-	err = svc.k8sClient.UpdateCR(testCtx, &volumeCR)
-	err = svc.ExpandVolume(testCtx, &volumeCR, capacity)
+	assert.NotNil(t, svc.k8sClient.UpdateCR(testCtx, &volumeCR))
+	err := svc.ExpandVolume(testCtx, &volumeCR, capacity)
 	assert.NotNil(t, err)
 	assert.Equal(t, codes.Internal, status.Code(err))
 
-	//Required capacity is more than capacity of AC
-	volAC := testAC2
+	// Required capacity is more than capacity of AC
+	volAC := &accrd.AvailableCapacity{
+		TypeMeta:   k8smetav1.TypeMeta{Kind: "AvailableCapacity", APIVersion: apiV1.APIV1Version},
+		ObjectMeta: k8smetav1.ObjectMeta{Name: uuid.New().String(), Namespace: testNS},
+		Spec: api.AvailableCapacity{
+			Size:         10000,
+			StorageClass: apiV1.StorageClassSystemLVG,
+		},
+	}
 	volAC.Spec.Location = testDrive1UUID
-	err = svc.k8sClient.CreateCR(testCtx, testAC2Name, &volAC)
-	assert.Nil(t, err)
+	assert.Nil(t, svc.k8sClient.CreateCR(testCtx, "", volAC))
 	err = svc.ExpandVolume(testCtx, &volumeCR, capacity)
 	assert.NotNil(t, err)
 	assert.Equal(t, codes.OutOfRange, status.Code(err))
 }
 
 func TestVolumeOperationsImpl_UpdateCRsAfterVolumeExpansion(t *testing.T) {
 	var (
-		svc *VolumeOperationsImpl
-		err error
+		svc      = setupVOOperationsTest(t)
+		volumeCR = testVolume1.DeepCopy()
+		err      error
 	)
 
-	volumeCR := testVolume1
-	svc = setupVOOperationsTest(t)
-	svc.cache.Set(volumeCR.Spec.Id, testNS)
+	svc.cache.Set(volumeCR.Spec.Id, volumeCR.Namespace)
 	volumeCR.Spec.CSIStatus = apiV1.Failed
-	err = svc.k8sClient.CreateCR(testCtx, volumeCR.Spec.Id, &volumeCR)
-	volAC := testAC2
+	err = svc.k8sClient.CreateCR(testCtx, volumeCR.Spec.Id, volumeCR)
+	volAC := &accrd.AvailableCapacity{
+		TypeMeta:   k8smetav1.TypeMeta{Kind: "AvailableCapacity", APIVersion: apiV1.APIV1Version},
+		ObjectMeta: k8smetav1.ObjectMeta{Name: uuid.New().String(), Namespace: testNS},
```

Comment:
AC is cluster-scoped. Do we need to set NS here?

Reply:
👍